app/models/ability.rb

```diff
@@ -2,15 +2,8 @@ class Ability
   include CanCan::Ability
 
   def initialize(user)
-    can :read, Post
-
-    return unless user.present?
-
-    can :manage, Post, author_id: user.id
-    can :manage, Comment, author_id: user.id
-
-    return unless user.role == 'admin'
-
-    can :manage, :all
+    can :manage, :all if user.role == 'admin'
+    can :destroy, Post, author: user
+    can :destroy, Comment, author: user
   end
 end
```

Comment:
- [ ] Kindly make sure a `User` is able to create a `Post` as right now your application raises the below error because of the authorizations rules. I suggest you update your `app/models/ability.rb` by referring this **[template](https://gist.github.com/tresorsawasawa/b7650b35420a9d5cbf841160dc8ffd98)**
![Screenshot from 2023-02-10 16-42-35](https://user-images.githubusercontent.com/83042742/218119663-e208490d-f542-4a52-b638-9141fc1e5c9a.png)